packages/beacon-node/src/network/util.ts

```diff
@@ -81,3 +84,7 @@ export function getConnection(connectionManager: ConnectionManager, peerIdStr: s
 export function isPublishToZeroPeersError(e: Error): boolean {
   return e.message.includes("PublishError.InsufficientPeers");
 }
+
+export function getDefaultDialer(libp2p: Libp2p): DefaultDialer {
+  return ((libp2p as unknown) as {components: Components}).components.dialer as DefaultDialer;
```

Comment:
looks like a work-around if libp2p doesn't public the DefaultDialer
@wemeetagain do you know if latest libp2p version has an official api to get this?

Reply:
There's no 'official' way to get this. They only make the `Dialer` interface public. Might be nice to expose the type in a nicer way :shrug: